### DISCUSSION TITLE
2D elastic model and free slip boundary conditions

### DISCUSSION THREAD

**Discussion by SebfraOslo** [kinnala/scikit-fem#936] [Q&A] [answered]
Hi,

I am writing a 2D elastic code. The model box is a square. I want to impose free slip boundary conditions on the top boundary (normal component 'u^2' = 0 and tangential component 'u^1' non defined).
However, it looks like the code understands that tangential component 'u^1' is qual to 0 too. 
Am I doing something wrong?
This is the part of my code: 

"mesh = (MeshQuad
        .init_tensor(np.linspace(0, Lx, nx), np.linspace(0, Ly, ny))
        .with_boundaries({
            'left': lambda x: x[0] == 0,
            'right': lambda x: x[0] == Lx,
            'bottom': lambda x: x[1] == 0,
            'top': lambda x: x[1] == Ly
        }))

e1 = ElementQuad1()
e = ElementVector(e1,2)
mapping = MappingIsoparametric(mesh, e1)
basis = Basis(mesh, e, mapping, 2)

# Boundary conditions
u = basis.zeros()
u[basis.get_dofs({'left'}).nodal['u^1']] = 0.
u[basis.get_dofs({'right'}).nodal['u^1']] = Vx
u[basis.get_dofs({'bottom'}).nodal['u^2']] = 0.
u[basis.get_dofs({'top'}).nodal['u^2']] = 0.

D = basis.get_dofs({'left','right','bottom','top'})

lambda_, mu = lame_parameters(E, nu)
K = asm(linear_elasticity(lambda_, mu), basis)

u = solve(*condense(K, x=u, D=D))"

Thank you,
Sebastien

**Comment by kinnala** [comment, marked as answer]
Passing
```python
D = basis.get_dofs({'left','right','bottom','top'})
```
to `condense(..., D=D)` causes all DOFs to be eliminated.

I suggest you do something like
```python
D1 = basis.get_dofs({'left', 'right', 'bottom'}).all()  # all DOFs fixed
D2 = basis.get_dofs('top').all('u^2')  # all 'u^2' DOFs fixed
D = np.concatenate((D1, D2))
```
Hopefully this solves your problem.

**Reply by SebfraOslo** [reply]
It is working nicely now. Thank you

Sebastien
